### ISSUE TITLE
Log file spammed with `[ERROR] [GLES2] GL_INVALID_OPERATION in glUniform1(uniform "proj"@0 is matrix)`

### ISSUE THREAD

**Issue by q234rty** [hyprwm/Hyprland#1720]
Steps to reproduce:

1. Run `Hyprland 2> error.log`
2. Inside the session run `tail -f error.log` in a terminal
3. Observe that the log file is spammed with `[ERROR] [GLES2] GL_INVALID_OPERATION in glUniform1(uniform "proj"@0 is matrix)`

**Comment by vaxerski** [comment]
![image](https://user-images.githubusercontent.com/43317083/223183773-928082bd-1b24-403b-89aa-3141a4b9b706.png)

Can't repro. Are you on git head?

**Comment by q234rty** [comment]
```
➜ q234rty ~ hyprctl version
Hyprland, built from branch main at commit 4f647a8e8be78d3817f126e41940bef49e81232edirty (fix mfact not applying to addmaster (1715)).
flags: (if any)
```

Also the error is not printed by Hyprland (it's absent in `hyprland.log` in `/tmp`), it's printed by something else (mesa?) to `stderr` (not `stdout`).

**Comment by vaxerski** [comment]
I am looking at stderr and stdout. Odd.

**Comment by q234rty** [comment]
Could you try running with `env MESA_DEBUG=1`? I don't have this variable set locally but may be mesa decides to print debug logs anyway for some reason.

**Comment by droc12345** [comment]
That should be a mesa error, and you're seeing msgs because you asked for them by setting MESA_DEBUG. 
Not sure it has anything to do with hyprland, other than it uses mesa.

**Comment by vaxerski** [comment]
it still shouldn't happen, it's an invalid op...

**Comment by q234rty** [comment]
>  Not sure it has anything to do with hyprland, other than it uses mesa.

It uses mesa in a wrong way.

**Comment by droc12345** [comment]
It's all in src/render/OpenGL.cpp, glUniform1f and glUniform1i

**Comment by q234rty** [comment]
Did a manual bisect and the first bad commit is https://github.com/hyprwm/Hyprland/commit/de5f1b2a833a59d3b0375f24675d27cea99e56d1

**Comment by vaxerski** [comment]
do you use a screen shader?

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/10901885/patch.txt)

Try this patch

**Comment by q234rty** [comment]
I can confirm that the first part of the patch fixed the issue for me. (The second part that changed the wlroots subproject commit doesn't seem to make much sense so I just removed it.)

**Comment by vaxerski** [comment]
it's an annoying artifact from my local clone.

**Comment by vaxerski** [comment]
fixed in 0aeb61a95af38ad20dd4d0f8a4ec1d485d73352f

**Comment by kragebein** [comment]
Awesome, i was trying to figure out why my logs were behaving like this. Updated.
